working/v3.0.0-rc2/payment-initiation-nz-openapi.yaml

```
@@ -0,0 +1,1596 @@
+                    $ref: "#/components/schemas/DomesticPaymentDebtorAccountResponse"
+                  Links:
+                    $ref: "#/components/schemas/Links"
+                  Meta:
+                    $ref: "#/components/schemas/Meta"
+                required:
+                  - Data
+                  - Links
+                  - Meta
+                additionalProperties: false
+        "400":
+          $ref: "#/components/responses/400ErrorResponse"
+        "401":
+          $ref: "#/components/responses/401ErrorResponse"
+        "403":
+          $ref: "#/components/responses/403ErrorResponse"
+        "405":
+          $ref: "#/components/responses/405ErrorResponse"
+        "406":
+          $ref: "#/components/responses/406ErrorResponse"
+        "429":
+          $ref: "#/components/responses/429ErrorResponse"
+        "500":
+          $ref: "#/components/responses/500ErrorResponse"
+        "501":
+          $ref: "#/components/responses/501ErrorResponse"
+        "503":
+          $ref: "#/components/responses/503ErrorResponse"
+      security:
+        - ThirdPartyOAuth2Security:
+            - payments
+tags:
+  - name: Domestic Payment Consents
+    description: Short lived payment-order consent
+  - name: Domestic Payments
+    description: Single, domestic, electronic credit payment-order
+  - name: Enduring Payment Consents
+    description: Long lived payment-order consent
+servers:
+  - url: https://api.provider.co.nz/open-banking-nz/v3.0
+components:
+  parameters:
+    x-idempotency-key-Param:
+      name: x-idempotency-key
+      in: header
+      description: Every request will be processed only once per
+        x-idempotency-key.  The Idempotency Key will be valid for 24 hours.
+      required: true
+      schema:
+        type: string
+        maxLength: 40
+        pattern: ^(?!\s)(.*)(\S)$
+    x-fapi-customer-ip-address-Param:
+      in: header
+      name: x-fapi-customer-ip-address
+      required: false
+      description: The Customer's IP address if the Customer is currently logged
+        in  with the Third Party.
+      schema:
+        type: string
+        pattern: ^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$
+    x-fapi-interaction-id-Param:
+      in: header
+      name: x-fapi-interaction-id
+      required: false
+      description: An RFC4122 UID used as a correlation id.
+      schema:
+        type: string
+    x-fapi-auth-date-Param:
+      in: header
+      name: x-fapi-auth-date
+      required: false
+      description: "The time when the Customer last logged in with the Third
+        Party.  All dates in the HTTP headers are represented as RFC 7231 Full
+        Dates. An example is below:  Sun, 10 Sep 2017 19:43:31 UTC"
+      schema:
+        type: string
+        pattern: ^(Mon|Tue|Wed|Thu|Fri|Sat|Sun), \d{2}
+          (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) \d{4}
+          \d{2}:\d{2}:\d{2} (GMT|UTC)$
```

Without the `>-` character, schema parsers will interpret newlines literally
```suggestion
        pattern: >-
          ^(Mon|Tue|Wed|Thu|Fri|Sat|Sun), \d{2}
          (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) \d{4}
          \d{2}:\d{2}:\d{2} (GMT|UTC)$
```

Without this change, a parser will translate the regex to the following:
`^(Mon|Tue|Wed|Thu|Fri|Sat|Sun), \\d{2}\n(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) \\d{4}\n\\d{2}:\\d{2}:\\d{2} (GMT|UTC)$`

Noting that the yaml newline character `\n` are being injected in after `\\d{2}` and `\\d{4}`

Note that similar parsing issues will occur with multi-lined yaml `description` blocks

---

That's not quite correct.  The YAML referenced above is using the [flow scalar plain style](https://yaml.org/spec/1.2-old/spec.html#id2788859).  Newlines are only inserted when there are no characters other than a newline, for example:

```yaml
key: this is a 
  long value
```

`key`'s corresponding value is interpreted as `this is a long value`.

However the following:
```yaml
key: this is a 

  long value
```

would be interpreted as: 
```
this is a
long value
```

So the regex value should parse as expected, and if not it is likely not an issue with formatting but the interpreter.  I've tested this in Java, Go, Python and NodeJS, and all correctly interpret the regular expression.

You can also export the YAML as JSON and confirm that the regular expression is correct.

The same applies to the `description` blocks.

I'm closing this conversation with no changes to the PR at this time.  Please raise a specification issue on the [API Centre JIRA](https://paymentsnz.atlassian.net/servicedesk/customer/portal/1/group/7) if you still believe this is an issue and we can review.